ReactApp/src/misc/api-fetch-setup.js

```diff
@@ -110,12 +89,30 @@ function corsMiddleware(options, next) {
 	return next(options);
 }
 
-function createHeadersMiddleware(authToken) {
+function apiPathModifierMiddleware(options, next) {
+	const { siteApiNamespace } = getGBKit();
+
+	if (siteApiNamespace) {
+		const originalPath = options.path;
+
+		// Determine the initial segments of the path where the namespace should be inserted.
+		// E.g /wp/v2/, /oembed/1.0/
+		const initialPathSegments =
+			originalPath.split('/').splice(1, 2).join('/') + '/';
+		const remainingPath = originalPath.replace(initialPathSegments, '');
+
+		// Update the path with the new namespace.
+		options.path = `${initialPathSegments}${siteApiNamespace}${remainingPath}`;
```

Comment:
This approach is far better than my original idea of passing a function from the host app. Nicely done! 👏🏻 

This is undoubtedly subjective, but I find the string splitting a bit difficult to comprehend. I feel a regex replace is simpler, but that may just be my familiarity with it. I recognize not everyone feels comfortable reading regex. 

WDYT? 

```suggestion
		// Insert the API namespace after the first two path segments. 
		options.path = options.path.replace(/^((?:\/[\w.-]+){2})/, `$1/${siteApiNamespace}`);
```

Reply:
Oooh I like this, thank you for suggesting that!